sips/mev.md

```diff
@@ -0,0 +1,91 @@
+| Author                    | Title                           | Category | Status              |
+| ------------------------- | ------------------------------- | -------- | ------------------- |
+| Moshe Revah (@moshe-blox) | Support externally built blocks | Core     | open-for-discussion |
+
+## Summary
+
+Support the production and proposal of blocks built by external entities, allowing operators to access blocks from a marketplace of builders such as [Flashbots](https://boost-relay.flashbots.net/).
+
+## Rational & Design Goals
+
+Block building is a computationally intensive task, and even more so when blockspace utilization and profit maximization are in mind. Additionally, many of the more profitable transactions, so-called MEV, are private and only accessible to validators through external block builders.
+
+Currently, operators wishing to offer competitive returns must be sophisticated block builders with private transaction flows. This SIP aims to change that by allowing them to register with and propose blocks from external builders who comply to [ethereum/builder-specs](https://github.com/ethereum/builder-specs).
+
+## Specification
+
+### Configuration
+
+Operators who wish to propose blinded blocks must configure their node to do so, otherwise it would only propose standard blocks and reject QBFT proposals containing blinded blocks.
+
+Therefore, validators should solely select operators with matching configuration, otherwise consensus might be unreachable.
+
+### Validator registration
+
+Builders require validators to publish a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) to set their `fee_recipient` and `gas_limit` preferences.
+
+When building a block for a validator, builders refer only to the registration with the highest timestamp.
+
+In the wild, some Ethereum validator clients currently publish this message every epoch.
+
+#### Fee recipients
+
+Validators may set their preferred `fee_receipient` address by calling `setFeeRecipientAddress` in the `SSVNetwork` contract. Validators may repeat this call as their preference changes over time.
+
+#### Signing
+
+At the start of every slot, operators select their active validators with `ShouldRegisterValidatorAtSlot`, and for those, produce a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) with their preferred `fee_recipient`.
+
+```go
+ValidatorRegistrationSlotInterval = 10 * SlotsPerEpoch
+
+func ShouldRegisterValidatorAtSlot(index phase0.ValidatorIndex, slot phase0.Slot) bool {
+    return (index + slot) % ValidatorRegistrationSlotInterval == 0
+}
+```
+
+#### Publishing
+
+Every slot, operators should publish the most recent successfully signed registration for `1/SlotsPerEpoch` of their validators, so that all validators are registered at least once per epoch.
```

Comment:
So let say that the most recent signed registration is at slot X, then operators will publish the same registration at X+1, X+2 and so on?

Is it neccessary?

Reply:
Regardless of when was the last signed registration, you submit for each validator once an epoch (the next portion of 1/32 at every slot). This SIP suggests to split the submissions per-slot, instead of a big one at the start/end of the epoch. Do you think I should specify exactly how to split your validators per-slot in this SIP? (for example with `(validatorIndex + slot) % 32 == 0`)

Why to re-submit the same signature?
1. A new relay may have been added and hasn't seen your registration yet
2. An existing relay forgot your registration for whatever reason
3. The Beacon node **might** be programmed to use the fee recipient from your registration when falling back to build local blinded blocks, but it forgets your registration after restarts

Prysm & Lighthouse seem to only sign once (and never again until preferences changed or node restarts), and publish once at the start of every epoch.

Ideally, we'd schedule signatures around `ValidatorAddedEvent` and `FeeRecipientAddressUpdated` to avoid re-signing when nothing has changed, or first signing too late (after a block proposal). However, we don't think it's too bad, and would like to keep that for a future PR.

Reply:
Rewritten this:

> Every epoch, operators should publish the most recent successfully signed registration for their entire validator set.
>
> This SIP suggests to spread registrations between slots. For example, at every slot, operators may publish for a fraction of their validators (corresponding to `1/SlotsPerEpoch`), ensuring a more evenly distributed load on the Beacon node and builder(s).